### PR TITLE
Fix issue when trying to give items with spaces in the name via player profile page

### PR DESCRIPTION
**Pull request by jcardoz**
This PR aims to fix: https://github.com/CatalysmsServerManager/7-days-to-die-server-manager/issues/324

* added the double-quotes
* added an item with a space to ensure that the test case passes in such a scenario

### CODE REVIEW

**Comment by halkeye** [comment]
Looks really good, but the fact that you changed the code, and the test, and nothing failed, says the test isn't really testing anything (not your fault, old test).

Can you update the test to check the spy (calledWith like you did in the other PR)

Also, then add a new test that has someone already used double quotes for an item name

**Comment by jcardoz** [comment]
> Looks really good, but the fact that you changed the code, and the test, and nothing failed, says the test isn't really testing anything (not your fault, old test).
> 
> Can you update the test to check the spy (calledWith like you did in the other PR)
> 
> Also, then add a new test that has someone already used double quotes for an item name

@halkeye 
can you be more specific in what you need added

test/integration/controllers/shop.test.js - does not have any calledWith references. Also the other PR #294 does not include  this type of test.

For the last comment - do i need to refer to `test/unit/helpers/sdtd/validate-item-name.test.js`?

**Comment by halkeye** [comment]
Right now test/integration/controllers/give-item.test.js just tests to see if the controller returned 200
it'll return 200 unless you do something really really weird (which because https://github.com/CatalysmsServerManager/7-days-to-die-server-manager/blob/master/test/integration/controllers/give-item.test.js#L6-L10 always returns success, it won't ever fail)

So i'm saying fix that first test so it checks the executeConsoleCommand stub defined in 6-10 and make sure its called how we want it to be called

Then add a new test where someone "gives" an item that already has quotes in it, so `itemName: '"something with spaces"'` because I suspect people are doing that right now to get around the space issue.

**Comment by jcardoz** [comment]
@halkeye - sorry to be dumb but just confirming from your comment I got:
* update the stub response for the first test - is there some file i can refer to an updated stub response
* add a new test for an item with spacing

Hope this covers it

**Comment by niekcandaele** [comment]
The problem is that the test doesn't really test anything right now. What @halkeye is saying is create a test that inputs `item name` and then check if the command was executed with `"item name"`. That's how we know it should be.

Then finally, because people are likely working around this bug right now and we do not want to break their configs: add a test where the input is `"item name"` and check if the output is still OK

Does that make it clearer @jcardoz ?

File: test/integration/controllers/give-item.test.js [hunk 1]
```diff
@@ -14,15 +14,35 @@ describe('/api/player/giveitem', function () {
       .post('/api/player/giveitem')
       .send({
         playerId: 1,
-        itemName: 'something',
+        itemName: 'ammo9mmBulletBall',
         amount: '1'
       });
-
+    console.log(response);
```
Comment (halkeye) [inline test/integration/controllers/give-item.test.js:20]:
```suggestion
```

^_^

File: test/integration/controllers/give-item.test.js [hunk 2]
```diff
@@ -14,15 +14,35 @@ describe('/api/player/giveitem', function () {
...
+    sandbox.stub(sails.helpers.sdtd, 'checkCpmVersion').callsFake(async () => 7);
+
+    const response = await supertest(sails.hooks.http.app)
+      .post('/api/player/giveitem')
+      .send({
+        playerId: 1,
+        itemName: 'doesnt exist',
+        amount: '1'
+      });
+    console.log(response);
```
Comment (halkeye) [inline test/integration/controllers/give-item.test.js:42]:
```suggestion
```

**Comment by halkeye** [comment]
Oh hey, I'm really sorry for taking over the PR, I normally don't like to do it, but after a few attempts to explain what I meant, I wanted to show the example tests, then got sidetracked adding a bunch more tests. Plus I thought it was someone on the discord that I knew better.

Hope thats okay, and that you can still learn from it.

**Review by niekcandaele — APPROVED** [review]

**Comment by jcardoz** [comment]
@halkeye - completely understand - i was not sure about the test cases and was trying and failing to get this working. Thank you for stepping in and fixing it.

**Pull request merged**